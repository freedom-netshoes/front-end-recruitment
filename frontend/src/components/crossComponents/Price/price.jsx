import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import { PriceFormat } from '../../../resources/helpers.jsx'
import CSSModules from 'react-css-modules'

import styles from './price.less'

@CSSModules(styles)
export default class Price extends Component {
  static defaultProps = {
    value: 0,
    type: 'full'
  };

  static propTypes = {
    value: PropTypes.number,
    type: PropTypes.oneOf(['full', 'separated'])
  };

  render () {
    const priceFormatted = PriceFormat(this.props.value)
    if (this.props.type === 'separated') {
      return (
        <span className='price'>
          <span className='price-symbol'>{priceFormatted.currencySymbol}</span>
          <span className='price-integer'>{priceFormatted.integer}</span>
          <span className='price-separator'>{priceFormatted.decimalSeparator}</span>
          <span className='price-decimal'>{priceFormatted.decimal}</span>
        </span>
      )
    }
    return (
      <span className='price'>
        {priceFormatted.toStringPrice}
      </span>
    )
  }
}
