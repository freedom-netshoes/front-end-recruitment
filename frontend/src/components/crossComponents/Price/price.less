@import '../../../styles/_less-variables.less';
@import '../../../styles/_less-mixins.less';
:global {
    .price {
        font-size: 30px;
    }
    .price-integer {
        .fontBold();
    }
    #cart-shop {
        color: @gold;
    }
    #products {
        .price-symbol,
        .price-separator,
        .price-decimal {
            font-size: 20px
        }

    }
}
