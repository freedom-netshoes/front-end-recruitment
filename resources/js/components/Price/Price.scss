@import '../../../scss/shared';

.price {
  color     : $yellow;
  font-size : 14px;
  margin-top: 10px;

  strong {
    @include open-sans-bold();
  }
}

:global(#products) .price {
  color     : $gray;
  font-size : 12px;
  margin-top: 8px;

  strong {
    @include open-sans-bold();
    font-size: 24px;
    position : relative;

    &:before {
      content         : "";
      background-color: $yellow;
      display         : block;
      height          : 2px;
      left            : 50%;
      margin-left     : -7.5px;
      position        : absolute;
      top             : 0;
      width           : 15px;
    }
  }
}

:global(#cart) {

  :global(.delete) .price {
    @include opacity(0.2);
    text-decoration: line-through;
  }

  :global(.conclude) .price {
    font-size: 24px;
  }

  .price {
    @include transition();
    color     : $yellow;
    font-size : 14px;
    margin-top: 10px;

    strong {
      @include open-sans-bold();
    }
  }
}