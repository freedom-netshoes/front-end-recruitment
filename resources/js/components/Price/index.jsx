import React, {Component, PropTypes} from 'react';
import CSSModules from 'react-css-modules';
import styles from './Price.scss'

@CSSModules(styles)
class Price extends Component {

  static defaultProps = {

  };

  static propTypes = {

  };

  render () {
    return (
      <p styleName='price'>R$ <strong>149</strong>,90</p>
    );
  }
}

export default Price;
