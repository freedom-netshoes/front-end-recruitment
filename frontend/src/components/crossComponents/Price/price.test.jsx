/* global describe, it, expect */
import React from 'react'
import Price from './price'
import { shallow } from 'enzyme'

describe('Testing <Price /> component', () => {
  it('renders the Price component', () => {
    const component = shallow(<Price />)
    expect(component).toHaveLength(1)
  })

  it('Testing <Price /> snapshop [Full]', () => {
    const component = shallow(<Price {...{
      value: 0,
      type: 'full'
    }} />)
    expect(component).toMatchSnapshot()
  })

  it('Testing <Price /> snapshop [Separated]', () => {
    const component = shallow(<Price {...{
      value: 0,
      type: 'separated'
    }} />)
    expect(component).toMatchSnapshot()
  })
})
