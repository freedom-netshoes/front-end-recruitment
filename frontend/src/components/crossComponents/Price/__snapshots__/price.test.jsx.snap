// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Price /> component Testing <Price /> snapshop [Full] 1`] = `
<span
  className="price"
>
  R$ 0.00
</span>
`;

exports[`Testing <Price /> component Testing <Price /> snapshop [Separated] 1`] = `
<span
  className="price"
>
  <span
    className="price-symbol"
  >
    R$ 
  </span>
  <span
    className="price-integer"
  >
    0.00
  </span>
  <span
    className="price-separator"
  />
  <span
    className="price-decimal"
  />
</span>
`;
